### PR TITLE
Regex based stack matching

### LINKED ISSUE #185: search based on stack version and not stack name
usecase: i have currently multiple stack versions running with the same version, they all a part of "the big stack", e.g.:
- kafka
- the rest api for incoming events
- a consumer
- zookeeper
- the management interface for kafka

to get a faster overview on e.g. all stacks or instances i could do 
```
senza instances | grep <stack_version>
```
but i think its more conventient maybe to have a command in senza built in to look for stacks having the same version. what do you think?

something like this (only a proposal, idk if this is e.g. backward-compatible):
```
senza instances --version=<stack_version>
```

### PR DESCRIPTION
**Pull request by jmcs**
This mostly fixes #185 but we still need documentation after.

### CODE REVIEW

File: senza/aws.py [hunk 1]
```diff
@@ -255,12 +256,22 @@ def matches_any(cf_stack_name: str, stack_refs: list):
...
+    except ValueError:
+        name = cf_stack_name
+        version = None
+
     for ref in stack_refs:
-        if ref.version and cf_stack_name == ref.cf_stack_name():
-            return True
-        elif not ref.version and (cf_stack_name or '').rsplit('-', 1)[0] == ref.name:
-            return True
+        matches_name = re.match(ref.name + '$', name)
```
Comment (rafaelcaricio) [inline senza/aws.py]: Why to append `'$'` to the regex?

Comment (jmcs) [inline senza/aws.py]: So searching for `my-app` doesn't automatically match `my-app-db` for example

Comment (rafaelcaricio) [inline senza/aws.py]: Then it is a problem of the regex. But I see your point.

**Comment by rafaelcaricio** [comment]
👍

**Comment by rafaelcaricio** [comment]
Needs rebase.

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**